### PR TITLE
Thomas style refactors

### PR DESCRIPTION
**Pull request by ThomasvLingen**
This PR contains refactors for all the Engine components, since a lot of issues actually got through the reviewing process (BOO ME)

### CODE REVIEW

File: src/SDLFacade.cpp [hunk 1]
```diff
@@ -64,81 +65,86 @@ namespace Engine {
...
+            SDL_DestroyRenderer(this->_renderer);
             return false;
         }
     }
 
-    bool SDLFacade::_init_window() {
-        _window = SDL_CreateWindow("Vidya game", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 500,
+    bool SDLFacade::_init_window()
+    {
+        this->_window = SDL_CreateWindow("Vidya game", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, 500,
```
Comment (jorgdebont) [inline src/SDLFacade.cpp]: vidya game should be placed in a string constant or define

File: src/SDLFacade.cpp [hunk 2]
```diff
@@ -64,81 +65,86 @@ namespace Engine {
...
     }
 
-    void SDLFacade::draw_line(const CoordinateDouble &position1, const CoordinateDouble &position2, const Color &color) {
-
-        SDL_SetRenderDrawColor(_renderer, color.r_value, color.g_value, color.b_value, 255);
-        SDL_RenderDrawLine(_renderer, position1.x, position1.y, position2.x, position2.y);
-
+    void SDLFacade::draw_line(const CoordinateDouble& position1, const CoordinateDouble& position2, const Color& color)
+    {
+        SDL_SetRenderDrawColor(this->_renderer, color.r_value, color.g_value, color.b_value, 255);
```
Comment (jorgdebont) [inline src/SDLFacade.cpp]: 255 is magic number

File: src/SDLFacade.cpp (hunk 1 above)
Comment (jorgdebont) [inline src/SDLFacade.cpp]: vidya game should be placed in a string constant or define

File: src/SDLFacade.cpp [hunk 3]
```diff
@@ -64,81 +65,86 @@ namespace Engine {
...
 
-    PressedKeys SDLFacade::get_keys() const {
+    PressedKeys SDLFacade::get_keys() const
+    {
         //todo
         return this->_keys_down;
     }
 
-    bool SDLFacade::draw_text(const string &text, const FontType &font, const Color &color, const CoordinateDouble &position) const {
+    bool SDLFacade::draw_text(const string& text, const FontType& font, const Color& color, const CoordinateDouble& position) const
```
Comment (jorgdebont) [inline src/SDLFacade.cpp]: perhaps split between lines, is over 70 chars i think

File: src/SDLFacade.cpp [hunk 4]
```diff
@@ -149,83 +155,92 @@ namespace Engine {
             SDL_FreeSurface(text_surface);
 
             return true;
-        } catch (int exception){
+        } catch (int exception) {
             TTF_CloseFont(render_font);
```
Comment (jorgdebont) [inline src/SDLFacade.cpp]: add todo, smart pointers

File: src/SDLFacade.cpp [hunk 5]
```diff
@@ -149,83 +155,92 @@ namespace Engine {
...
+    int SDLFacade::get_width() const
+    {
         return this->_width;
     }
 
-    bool SDLFacade::_init_fonts(){
-        if(!_load_font("res/alterebro_pixel.ttf", FontType::alterebro_pixel, 30)){
+    bool SDLFacade::_init_fonts()
+    {
+        if (!_load_font("res/alterebro_pixel.ttf", FontType::alterebro_pixel, 30)) {
```
Comment (jorgdebont) [inline src/SDLFacade.cpp]: kan one liner worden:
`return _load_font(res/alterebro_pixel.ttf", FontType::alterebro_pixel, 30);`

Comment (jorgdebont) [inline src/SDLFacade.cpp]: mochten er meerdere fonts komen, dan maak je daar aparte functie voor

File: src/SDLFacade.hpp [hunk 6]
```diff
@@ -61,29 +60,29 @@ namespace Engine {
...
+        SDLFacade(const function<void()>& callback_func);
 
         virtual ~SDLFacade();
 
         bool init();
 
         void clear_screen(); //draws the background
 
-        void draw_line(const CoordinateDouble &position1, const CoordinateDouble &position2, const Color &color);
+        void draw_line(const CoordinateDouble& position1, const CoordinateDouble& position2, const Color& color);
```
Comment (jorgdebont) [inline src/SDLFacade.hpp]: rename positions to start and end, or doesnt drawline work like that?

File: src/SDLFacade.cpp (hunk 1 above)
Comment (ThomasvLingen) [inline src/SDLFacade.cpp]: Is this a stylistic issue?

File: src/SDLFacade.cpp (hunk 2 above)
Comment (ThomasvLingen) [inline src/SDLFacade.cpp]: Is this a stylistic issue?

File: src/SDLFacade.cpp (hunk 4 above)
Comment (ThomasvLingen) [inline src/SDLFacade.cpp]: You can't do that for render_font, since you're not allowed to `delete` it.
You have to clean it up with the SDL_TTF function

**Comment by ThomasvLingen** [comment]
I'm only here for the stylistic issues, I'd say other stuff should be @WoutervanGeel or @Mfrielink 's responsibility.

You can make Issues for those problems!

File: src/SDLFacade.cpp (hunk 3 above)
Comment (ThomasvLingen) [inline src/SDLFacade.cpp]: Is our character limit per line 70 characters?

As far as I know, I have never specified this in the styleguide.

File: src/SDLFacade.hpp (hunk 6 above)
Comment (ThomasvLingen) [inline src/SDLFacade.hpp]: Done

File: src/SDLFacade.cpp (hunk 5 above)
Comment (ThomasvLingen) [inline src/SDLFacade.cpp]: Done

File: src/SDLFacade.cpp (hunk 3 above)
Comment (jorgdebont) [inline src/SDLFacade.cpp]: then why split it in other parts of the file where we went over 70

File: src/SDLFacade.cpp (hunk 4 above)
Comment (jorgdebont) [inline src/SDLFacade.cpp]: true

**Pull request merged**